kdc/krb5tgs.c

```diff
@@ -1113,6 +1105,12 @@ tgs_check_authenticator(krb5_context context,
 	krb5_free_error_message(context, msg);
 	goto out;
     }
+
+    /*
+     * according to RFC1510 it doesn't need to be keyed,
+     * but according to the latest draft it needs to.
+     */
```

Comment:
obviously this comment didn't make sense and still doesn't. RFC4120 doesn't say it needs to be keyed, just collision proof FWIW.

Reply:
Make sense since it is encrypted anyway, perhaps worth noting what RFC4120 says, the comment might have predated it, who knows :)

Reply:
Yeah, I was a bit confused by that. I checked with @nicowilliams and he said the authenticator checksum was more channel bindings, i.e. an extensibility point for protecting additional data (e.g. as used by GSS-API).

Reply:
That's when used in AP request to a server, in AP request to KDC aka TGS-REQ it uses regular keyed checksum afair.

Reply:
Right, but RFC4120 doesn't say it has to be keyed, only collision-proof. Section 3.3.2.

Reply:
Yeah, again make sense since it is in encrypted envelop, not sure what's the rational to have it to begin with, i'll look at the RFC..

Reply:
I'm actually going to introduce a new `_kdc_verify_checksum()` function that maps `KRB5_PROG_SUMTYPE_NOSUPP` to `KRB5KDC_ERR_SUMTYPE_NOSUPP` to increase our RFC4120 compliance.

Reply:
RFC4120 3.3.2 says the checksum is application specific. I don't even think MIT checks it tbh.

Reply:
Skimming the MIT code, it uses it only for GSS, and that's all within the Kerberos GSS mechanism itself (notwithstanding the machinery for `krb5_auth_con_set_checksum_func()`).